ase/sw/app_backend.c

```diff
@@ -45,9 +45,9 @@ struct buffer_t *mmio_region;
 // UMAS region
 struct buffer_t *umas_region;
 
-// Workspace metadata table
-struct wsmeta_t *wsmeta_head = (struct wsmeta_t *) NULL;
-struct wsmeta_t *wsmeta_end = (struct wsmeta_t *) NULL;
+// Record of MMIO and UMAS memory
+struct buffer_t *buf_head = (struct buffer_t *) NULL;
+struct buffer_t *buf_end = (struct buffer_t *) NULL;
```

Comment:
Can you elaborate on the reason for this change? Are you renaming wsmeta to buffer for consistency? Or is there already a buffer definition that can cover what wsmeta does, and you're consolidating?

I'm not saying it's not a good idea, I'd just like to understand your reasoning behind it. It's always a good idea to outline the "why" in your commit messages so that developers not familiar with the changes can understand the reasons for them.

You can also add comments to your on code in GitHub to explain changes if you think that's worthwhile for making the review go smoothly.

Reply:
I think that struct wsmeta_t definition is not needed: 
1. struct buffer_t has all the information needed to keep track of the buffers. Adding another layer of buffer definion with wsmeta_t make the code not readable. 
2. There were memory leak issues since all the memories allocate for wsmeta_t were not released.  
Please check the buffer_t and wsmeta_t definition below. Writing simple software code while achieving the same goal is better. 

// Buffer information structure --
//   Be careful of alignment within this structure!  The layout must be
//   identical on both 64 bit and 32 bit compilations.
struct buffer_t			//  Descriptiion                    Computed by
{				// --------------------------------------------
	int32_t index;		// Tracking id                     | INTERNAL
	int32_t valid;		// Valid buffer indicator          | INTERNAL
	uint64_t vbase;		// SW virtual address              |   APP
	uint64_t pbase;		// SIM virtual address             |   SIM
	uint64_t fake_paddr;	// unique low FPGA_ADDR_WIDTH addr |   SIM
	uint64_t fake_paddr_hi;	// unique hi FPGA_ADDR_WIDTH addr  |   SIM
	int32_t is_privmem;	// Flag memory as a private memory |
	int32_t is_mmiomap;	// Flag memory as CSR map          |
	int32_t is_umas;	// Flag memory as UMAS region      |
	uint32_t memsize;	// Memory size                     |   APP
	char memname[ASE_FILENAME_LEN];	// Shared memory name              | INTERNAL
	struct buffer_t *next;
};


/*
 * Workspace meta list
 */
struct wsmeta_t {
	int index;
	int valid;
	uint64_t *buf_structaddr;
	struct wsmeta_t *next;
};Leave a comment